src/test/java/com/fasterxml/jackson/databind/ser/enums/EnumSetSerializationTest.java

```diff
@@ -0,0 +1,69 @@
+package com.fasterxml.jackson.databind.ser.enums;
+
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper.DefaultTyping;
+import com.fasterxml.jackson.databind.jsontype.BasicPolymorphicTypeValidator;
+
+public class EnumSetSerializationTest extends BaseMapTest {
+
+	static enum MyEnum {
+		ITEM_A, ITEM_B;
+	}
+
+	static class EnumSetHolder {
+		private Set<MyEnum> enumSet; // use Set instead of EnumSet for type of this
+
+		public Set<MyEnum> getEnumSet() {
+			return enumSet;
+		}
+
+		public void setEnumSet(Set<MyEnum> enumSet) {
+			this.enumSet = enumSet;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (!(o instanceof EnumSetHolder)) {
+				return false;
+			}
+			EnumSetHolder eh = (EnumSetHolder) o;
+			if (eh == this) {
+				return true;
+			}
+			if (enumSet == null) {
+				if (eh.getEnumSet() == null) {
+					return true;
+				} else {
+					return false;
+				}
+			} else {
+				if (eh.getEnumSet() == null) {
+					return false;
+				}
+				return enumSet.containsAll(eh.getEnumSet());
+			}
+		}
+	}
+
+	public void testSerialization() throws Exception {
+		ObjectMapper mapper = jsonMapperBuilder().build()
+				.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
```

Comment:
Never disable this for unit tests (except when testing feature itself) -- it has a nasty habit of hiding real problems, and developers wrongly enable it out of habit (there are good reasons to enable it too, just not always).

Reply:
sure, will remove this configuration.